File: new/models/PetugasModel.js
import { Sequelize } from "sequelize";
import db from "../config/database.js";
import Tanggapan from "./TanggapanModel.js";

const {DataTypes} = Sequelize;

const Petugas = db.define('petugas',{
    id_petugas : {
        autoIncrement: true,
        type : DataTypes.INTEGER,
        primaryKey : true
    },
    nama_petugas : DataTypes.STRING,
    username : DataTypes.STRING,
    password : DataTypes.STRING,
    telp : DataTypes.STRING,
    level : DataTypes.STRING
},{
    freezeTableName : true
});



export default Petugas;

// (async()=>{
//     await db.sync({});
// }) ();

File: lib/petugas/model/petugas_model.dart
class PetugasModel{
  int? id_petugas;
  String? nama_petugas;
  String? username;
  String? password;
  String? telp;
  String? level;


  PetugasModel(this.id_petugas, this.nama_petugas, this.username, this.level);

  PetugasModel.fromJson(Map<String, dynamic> json) {
    id_petugas = json['id_petugas'];
    nama_petugas = json['nama_petugas'];
    username = json['username'];
    password = json['password'];
    telp = json['telp'];
    level = json['level'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id_petugas'] = this.id_petugas;
    data['nama_petugas'] = this.nama_petugas;
    data['username'] = this.username;
    data['level'] = this.level;
    return data;
  }
}